src/EventSourcedAggregateRootRepository.php

```diff
@@ -77,12 +77,17 @@ public function persist(object $aggregateRoot): void
         $this->persistEvents(
             $aggregateRoot->aggregateRootId(),
             $aggregateRoot->aggregateRootVersion(),
-            ...$aggregateRoot->releaseEvents()
+            $aggregateRoot->releaseEvents()
         );
     }
 
-    public function persistEvents(AggregateRootId $aggregateRootId, int $aggregateRootVersion, object ...$events): void
+    /**
+     * @param iterable<object> $events
+     */
+    public function persistEvents(AggregateRootId $aggregateRootId, int $aggregateRootVersion, iterable $events): void
     {
+        $events = [...$events];
```

Comment:
This will unpack the iterable fully to memory, right? I wonder if there is a way to achieve this by using only generators. I couldn't think on a straightforward way to do so.